### DISCUSSION TITLE
Is there a way to reduce padding size?

### DISCUSSION THREAD

**Discussion by thvdveld** [tock/tock#4146] [Q&A] [answered]
I'm trying to add multiple small (2048 bytes each) applications on the nRF52840, but I only can put 3 on it. The padding added between the applications seems way too big (first padding is 22528 bytes and the second one is 260096 bytes). I get the following messages when I try to install more applications:

```
[ERROR  ] Could not meet fixed address requirements.
[ERROR  ] Unable to find a valid sort order to flash apps.
```

Is there a way to reduce the padding size?

**Comment by bradjc** [comment, marked as answer]
Yes. You can add more locations here: https://github.com/tock/libtock-rs/blob/1dc6133f3b6f837ed71716daf4d96bdecfa3a28c/Makefile#L184-L186.

By default we pick some "reasonable" addresses given we don't know the actual size of what apps a user will want to flash on to a board. We also choose three different combinations (which is why you can only get three apps to install) as an arbitrary amount.

You will want to choose flash addresses which are closer together given you are building small apps. You will also need to set the memory addresses to be close enough to not run out of memory but also large enough the apps can be allocated enough memory. Finally, the flash and RAM addresses have to maintain ordering with other apps (i.e., if one app uses a larger flash start address it must also use a larger RAM start address as apps are allocated RAM in the order they are loaded in flash).
